dbms/src/TiDB/Decode/Vector.cpp

```diff
@@ -50,62 +51,75 @@ Float64 VectorFloat32Ref::l2SquaredDistance(VectorFloat32Ref b) const
 {
     checkDims(b);
 
-    Float32 distance = 0.0;
-    Float32 diff;
-
-    for (size_t i = 0, i_max = size(); i < i_max; ++i)
+    static simsimd_metric_punned_t metric = nullptr;
+    if (metric == nullptr)
```

Comment:
Can `VectorFloat32Ref::l2SquaredDistance` be called concurrently?

Reply:
Good catch! Fixed in https://github.com/pingcap/tiflash/pull/9372/commits/b427bee1104dfaada45901856124f2344777fc79